site/content/contribute/developer-setup/docker.md

```diff
@@ -0,0 +1,128 @@
+```
+
+## mysql
+
+From https://dev.mysql.com/doc/refman/8.3/en/introduction.html:
+
+> The MySQL software delivers a very fast, multithreaded, multi-user, and robust SQL (Structured Query Language) database server.
+
+This is an alternate database supported by Mattermost, but not recommended for new deployments.
+
+To use with Mattermost, be sure to configure the following settings:
+```
+MM_SQLSETTINGS_DRIVERNAME=mysql
+MM_SQLSETTINGS_DATASOURCE=mmuser:mostest@tcp(localhost:3306)/mattermost_test?charset=utf8mb4,utf8\u0026readTimeout=30s\u0026writeTimeout=30s
+```
+
+## inbucket
+
+From https://inbucket.org/about/:
+
+> Inbucket is an email testing application; it will accept messages for any email address and make them available to view via a web interface. If you've ever used mailinator.com, you already have a good idea of what Inbucket does. The benefit of Inbucket is that it is an application instead of a hosted service; you may run it on your own private network, or even your desktop. 
+
+Use this during development to "receive" email confirmations, password resets, or message notifications.
+
+To use with Mattermost, be sure to configure the following settings:
+```
+MM_EMAILSETTINGS_ENABLESMTPAUTH=false
+MM_EMAILSETTINGS_SMTPUSERNAME=
+MM_EMAILSETTINGS_SMTPPASSWORD=
+MM_EMAILSETTINGS_SMTPSERVER=localhost
+MM_EMAILSETTINGS_SMTPPORT=10025
+```
+
+When running, access the web interface at [http://localhost:9001/](http://localhost:9001/).
+![inbucket](/img/docker/inbucket.png)
+
+## grafana
+
+From https://grafana.com/docs/
+
+> Collect, correlate, and visualize data with beautiful dashboards using our open source data visualization and monitoring solution.
+
+Grafana is where all the metrics and logs collected by Prometheus, Loki and promtail come together. Panels visualize the data and are grouped into dashboards. The home dashboard links out to various performance dashboards, lists which Docker services are currently online, has quick links to various filtered log views, and panels showing the most recent Mattermost and Docker container logs.
+
+When running, access the web interface at [http://localhost:3000](http://localhost:3000).
+![grafana](/img/docker/grafana.png)
+
+## prometheus
+
+From https://prometheus.io/docs/introduction/overview/:
+
+> Prometheus collects and stores its metrics as time series data, i.e. metrics information is stored with the timestamp at which it was recorded, alongside optional key-value pairs called labels.
+
+Mattermost exposes metrics at [http://localhost:8067/metrics](http://localhost:8067/metrics) which are scraped periodically by Prometheus to form a time series database. While you can access Prometheus directly to view and graph this collected data, typically this is used in tandem with Grafana for a rich dashboard experience.
+
+To use with Mattermost, be sure to install a Mattermost enterprise license and configure the following settings:
```

Comment:
Can this be used without a license withing a dev environment?

Reply:
In the past, it was impossible (as we needed closed source code). Now, it's technically possible by making code changes now that it's source available, but it would be nice to go one step further and port over the Playbooks model of unlocking enterprise functionality using configuration setting that are unsafe in production. For now, I'm inclined to keep it documented as Enterprise only.

Reply:
> and port over the Playbooks model of unlocking enterprise functionality using configuration setting that are unsafe in production

Ah, this is what I had in mind! That was a great strategy in my opinion, hopefully we can add it to the main repo as well :crossed_fingers: